### DISCUSSION TITLE
How to customise post title header size?

### DISCUSSION THREAD

**Discussion by ilg-ul** [facebook/docusaurus#9666] [Q&A] [answered]
As per the [documentation](https://docusaurus.io/docs/next/styling-layout#global-styles) I tried to customise the post title size by adding definitions to the `custom.css` file, but apparently this file is included before the theme definitions, and the change is not effective.
 
I did the following:

```
npx create-docusaurus@3.0.1 website classic --typescript
```

Using the Safari Element Inspector, I copied the definitions from the styles.css displayed by Safari, updated the size and added the following lines at the end of `website/src/css/custom.css`:

```
/* Reduce title size */
.title_node_modules-\@docusaurus-theme-classic-lib-theme-BlogPostItem-Header-Title-styles-module {
  font-size: 2.5rem;
}
```

Using the Safari Element Inspector, now I see:

![Screenshot 2023-12-23 at 10 54 45](https://github.com/facebook/docusaurus/assets/362732/355c8fb2-710f-468f-9444-067d6eef2db1)

The custom definition with 2.5rem is shown at line 3112, while the other definition with 3rem is shown at line 3617, which is after the custom definition, so it has precedence.

Is there any way to customise the post title size?
 
I can provide a separate git project if the above are not enough to reproduce the issue.

**Comment by Josh-Cena** [comment]
First, do not use selectors that include `node_modules`: they are modularized and unstable. Please read https://docusaurus.io/docs/styling-layout#global-styles:

> CSS module class names. These class names have a hash in production (`codeBlockContainer_RIuc`) and are appended with a long file path in development. They are considered implementation details and you should almost always avoid targeting them in your custom CSS. If you must, you can use an attribute selector (`[class*='codeBlockContainer']`) that ignores the hash.

Your best option is to target the tag instead, such as:

> `.blog-list-page h2[class*="title"]`

**Comment by ilg-ul** [comment, marked as answer]
Thank you @Josh-Cena, that was helpful.

The final solution requires two definitions, one for the page with the blog list, and one for the page with the actual post:

```
/* Reduce title size in blog list */
.blog-list-page h2[class*="title"]
{
  font-size: 2.5rem;
}

/* Reduce title size in blog page */
.blog-post-page h1[class*="title"]
{
  font-size: 2.5rem;
}
```

I am a bit confused about the difference between these two, the blog list uses H2 for the title, while the blog post uses H1 for the same title.

Is this intentional?

**Reply by Josh-Cena** [reply]
Yes. A page should generally not have multiple H1s, so for the list we have to use all H2.

**Reply by ilg-ul** [reply]
Makes sense. Thank you.
